library/manageiq_provider.py

```diff
@@ -356,12 +359,19 @@ def generate_auth_key_config(self, role, authtype, hostname, port, token, provid
         if provider_ca_path:
             with open(provider_ca_path, 'r') as provider_ca_file:
                 provider_ca_content = provider_ca_file.read()
+        elif provider_ca_path == "":
+            provider_ca_content = ""
```

Comment:
I don't like this None / something else structure, with if is not None below.
Can you put `endpoint = {...` first, then do this if/elif setting `endpoint['endpoint']['certificate_authority']`?

Also don't call the var `endpoint` if it has 'endpoint' and 'authentication' inside.  How about `config`?